### PR TITLE
Issue 4796 - Add support for nsslapd-state to CLI & UI

### PR DESCRIPTION
**Pull request by mreynolds389**
Description:  

Add support for nsslapd-state to lib389 and UI.  Also added a check to prevent the changing of nsslapd-state for replicated suffixes.

Also did a little UI cleanup where a bottom margin was added  to the bottom of pages instead of using "hr" to create the gap.

relates: https://github.com/389ds/389-ds-base/issues/4796

### CODE REVIEW

**Comment by mreynolds389** [comment]
@jchapma - nice catch!!  Fixed, please review...

File: src/cockpit/389-console/src/lib/database/suffix.jsx [hunk 1]
```diff
@@ -773,6 +775,10 @@ export class Suffix extends React.Component {
             cmd.push("--dncache-memsize=" + this.state.dncachememsize);
             requireRestart = true;
         }
+        if (this.state._dbstate != this.state.dbstate) {
```
Comment (droideck) [inline src/cockpit/389-console/src/lib/database/suffix.jsx:778]: Currently, if the user changes 'Backend State' to 'Refferal' but no referrals exist, the user gets 'Unwilling to perform' message in the notification.

It is a bit cryptic.
Only in the Console Log we can see:

```
cockpit.js:1 {
    "msgtype": 103,
    "msgid": 9,
    "result": 53,
    "desc": "Server is unwilling to perform",
    "ctrls": [],
    "info": "need to set nsslapd-referral before moving to referral state\n"
}
```

I wonder if it's worth adding additional validation here (or print the meaningful message in the notification).

**Review by droideck — APPROVED** [review]
It's really minor (as we have a similar situation across other UI modules).
So you have my 'ack' as everything else is good!

File: src/cockpit/389-console/src/lib/database/suffix.jsx (hunk 1 above)
Comment (mreynolds389) [reply src/cockpit/389-console/src/lib/database/suffix.jsx:778]: We can add a check for "info" in the message and print it..  I've done this in other parts of the code, but it was too big of change to do it across the board with my previous work.  Maybe we should open a new ticket for adding the "info" check when doing error notifications

Comment (mreynolds389) [reply src/cockpit/389-console/src/lib/database/suffix.jsx:778]: Ok, I fixed it, and I will open a ticket to make this approach consistent across the UI

**Pull request merged**